src/formattedtext.jl

```diff
@@ -114,18 +121,30 @@ function Makie.plot!(plot::FormattedText{<:Tuple{<:Markdown.Paragraph}})
 
     plot.maxwidth[] = plot.maxwidth[]
 
-    text!(plot, glyphcollection; plot.attributes...)
-
-    plot
-end
+    txt = text!(plot, glyphcollection; plot.attributes...)
+
+    on(emojicollection) do ec
+        # txtbbox = Rect2f(boundingbox(txt))
+        # bbox = Rect2f(0,0,0,0)
+        # for bb in glyphbbs[]
+        #     display(bb.origin)
+        # end
+        for (shorthand, pos) in ec
+            filename = emoji_filename_png(shorthand)
+            e_img = load_emoji_image(filename)
+            bbox = glyphbbs[][pos]
+            w, h = width(bbox), height(bbox)
+            # these bboxes don't know about their predecessors, hence, the emojis are placed wrongly
+            box, boy = bbox.origin
+            msize = max(w, h)
+            scatter!(plot, (box, boy), marker=e_img, markersize=200, markerspace=:data,
+                     space=:data)
+        end
+    end
```

Comment:
This will not behave well whenever `emojicollection` changes, since the scatters are made individually.  I rewrote this to the form I used in MakieTeX for the same thing (beware that it won't work in CairoMakie, but that's simple to write):

```suggestion
    emoji_positions = Observable{Vector{Point2f}}(Point2f[(0,0)])
    emoji_images    = Observable{Matrix{Makie.Colors.RGBAf}}(rand(Makie.Colors.RGBAf, 1,1))
    emoji_sizes     = Observable{Vector{Vec2f}}(Vec2f[(0,0)])
    on(emojicollection) do ec
        # empty and size all arrays
        empty!(emoji_positions.val)
        empty!(emoji_images.val)
        empty!(emoji_sizes.val)
        sizehint!(emoji_positions, length(ec))
        sizehint!(emoji_images, length(ec))
        sizehint!(emoji_sizes, length(ec))
        
        for (shorthand, pos) in ec
            filename = emoji_filename_png(shorthand)
            bbox = glyphbbs[][pos]
            w, h = width(bbox), height(bbox)
            # these bboxes don't know about their predecessors, hence, the emojis are placed wrongly
            box, boy = bbox.origin
            # push the position vectors to all arrays
            push!(emoji_positions, Point2f(bbox.origin))
            push!(emoji_images, load_emoji_image(filename))
            push!(emoji_sizes, Vec2f(w, h))
        end
        # notify all observables in sequence
        notify(emoji_positions)
        notify(emoji_images)
        notify(emoji_sizes)
    end
    
    notify(glyphcollection_emojicollection)

    scatter!(
        plot, emoji_positions; 
        marker = emoji_images, 
        markersize = emoji_sizes,
        space = plot.space,
        markerspace = :pixel,# TODO: should be `plot.markerspace,`
        # This is required, and makes the position be at the bottom left of the marker,
        # as opposed to the center.  If this errors you may try setting is to 0.0 instead.
        marker_offset = Vec2f(0),
    )
```

This way, the old emojis are deleted and all emojis are replotted whenever the text changes.  It might be a bit of a hassle to layout, but this is actually not so difficult to optimize - if you factor the generation of images out, as done e.g. [in MakieTeX](https://github.com/JuliaPlots/MakieTeX.jl/blob/216d04f1129a08b8f904d7777ec2673d9f131f10/src/recipe.jl#L61-L104), then it becomes more efficient if you are actually just changing the layout as opposed to the actual text.

Another thing which might help is caching the loaded emojis in a Dict somewhere.

I haven't properly tested this code but it should give a general idea.

Reply:
> This will not behave well whenever emojicollection changes, since the scatters are made individually, ...

Yes, I know, this was what I referred to as rendering in a `on(emojicollection)` block does not work. I just added it for now since the `scatter` with Vectors gave me a `StackOverflow` for some reason.

I always thought operations like `empty!, push!` applied to `Observables` do trigger updates so I avoided them. If that is not the case than we should really do it this way.

One question: Having three `notify`s in this block will trigger the subsequent `scatter` three times or? Is there a way around this?

Reply:
I had to modify your suggestion a little since the `push!, empty!, sizehint` methods only work on the `val` member of `Observables. But fixing this brings back the `StackOverflow` error I mentioned earlier. Here is a MWE to reproduce:
```julia
using GLMakie
GLMakie.activate!()
pos = Point2f[(0,0)]
img = Matrix{Makie.RGBAf}[rand(Makie.RGBAf,1,1)]
scatter(pos, marker=img) # StackOverflow
scatter(pos, marker=img[1]) # works
```

I assume this is a `Makie` bug, or?

Reply:
I don't get any stackoverflow on master - the code runs perfectly, even with larger images as markers.  MakieTeX works in GLMakie as well, and it uses the same approach.

Could you post the actual error and your current GLMakie and Makie versions?

Reply:
Sorry, wrote this late at night, edited a bit in the morning but couldn't comment.  Yep, mutations/operations if done on the underlying object do not innately update the observable (only setindex does).  

The three notifys will trigger scatter three times.  Unfortunately there's no way around that, and it's pretty much expected behaviour since the GLMakie internal values will only update when the observables are updated.

Reply:
<details>
  <summary>Stack trace</summary>

```julia
julia> include("mwe_scatter.jl")
Error showing value of type Makie.FigureAxisPlot:
ERROR: StackOverflowError:
Stacktrace:
  [1] _collect(c::Vector{Matrix{ColorTypes.RGBA{Float32}}}, itr::Base.Generator{Vector{Matrix{ColorTypes.RGBA{Float32}}}, typeof(size)}, #unused#::Base.EltypeUnknown, isz::Base.HasShape{1})
    @ Base ./array.jl:0
  [2] collect_similar
    @ ./array.jl:653 [inlined]
  [3] map
    @ ./abstractarray.jl:2849 [inlined]
  [4] draw_scatter(p::Tuple{Observable{Vector{Matrix{ColorTypes.RGBA{Float32}}}}, Observable{Vector{Point{2, Float32}}}}, data::Dict)
    @ GLMakie.GLVisualize ~/.julia/packages/GLMakie/XG7Hm/src/GLVisualize/visualize/particles.jl:187
  [5] draw_scatter(p::Tuple{Observable{Vector{Matrix{ColorTypes.RGBA{Float32}}}}, Observable{Vector{Point{2, Float32}}}}, data::Dict) (repeats 9982 times)
    @ GLMakie.GLVisualize ~/.julia/packages/GLMakie/XG7Hm/src/GLVisualize/visualize/particles.jl:212
  [6] (::GLMakie.var"#125#129"{Scene, Scatter{Tuple{Vector{Point{2, Float32}}}}})(gl_attributes::Dict{Symbol, Any})
    @ GLMakie ~/.julia/packages/GLMakie/XG7Hm/src/drawing_primitives.jl:250
  [7] (::GLMakie.var"#107#110"{GLMakie.var"#125#129"{Scene, Scatter{Tuple{Vector{Point{2, Float32}}}}}, GLMakie.Screen, Scene, Scatter{Tuple{Vector{Point{2, Float32}}}}})()
    @ GLMakie ~/.julia/packages/GLMakie/XG7Hm/src/drawing_primitives.jl:107
  [8] get!(default::GLMakie.var"#107#110"{GLMakie.var"#125#129"{Scene, Scatter{Tuple{Vector{Point{2, Float32}}}}}, GLMakie.Screen, Scene, Scatter{Tuple{Vector{Point{2, Float32}}}}}, h::Dict{UInt64, GLMakie.GLAbstraction.RenderObject}, key::UInt64)
    @ Base ./dict.jl:464
  [9] cached_robj!(robj_func::GLMakie.var"#125#129"{Scene, Scatter{Tuple{Vector{Point{2, Float32}}}}}, screen::GLMakie.Screen, scene::Scene, x::Scatter{Tuple{Vector{Point{2, Float32}}}})
    @ GLMakie ~/.julia/packages/GLMakie/XG7Hm/src/drawing_primitives.jl:85
 [10] draw_atomic(screen::GLMakie.Screen, scene::Scene, x::Union{Scatter, MeshScatter})
    @ GLMakie ~/.julia/packages/GLMakie/XG7Hm/src/drawing_primitives.jl:207
 [11] insert!(screen::GLMakie.Screen, scene::Scene, x::Combined)
    @ GLMakie ~/.julia/packages/GLMakie/XG7Hm/src/drawing_primitives.jl:121
 [12] insertplots!(screen::GLMakie.Screen, scene::Scene)
    @ GLMakie ~/.julia/packages/GLMakie/XG7Hm/src/screen.jl:56
 [13] #40
    @ ~/.julia/packages/GLMakie/XG7Hm/src/screen.jl:58 [inlined]
 [14] foreach(f::GLMakie.var"#40#42"{GLMakie.Screen}, itr::Vector{Scene})
    @ Base ./abstractarray.jl:2694
 [15] insertplots!(screen::GLMakie.Screen, scene::Scene)
    @ GLMakie ~/.julia/packages/GLMakie/XG7Hm/src/screen.jl:58
 [16] #40
    @ ~/.julia/packages/GLMakie/XG7Hm/src/screen.jl:58 [inlined]
 [17] foreach(f::GLMakie.var"#40#42"{GLMakie.Screen}, itr::Vector{Scene})
    @ Base ./abstractarray.jl:2694
 [18] insertplots!(screen::GLMakie.Screen, scene::Scene)
    @ GLMakie ~/.julia/packages/GLMakie/XG7Hm/src/screen.jl:58
 [19] backend_display(screen::GLMakie.Screen, scene::Scene)
    @ GLMakie ~/.julia/packages/GLMakie/XG7Hm/src/display.jl:18
 [20] backend_display(#unused#::GLMakie.GLBackend, scene::Scene)
    @ GLMakie ~/.julia/packages/GLMakie/XG7Hm/src/display.jl:4
 [21] display(scene::Scene)
    @ Makie ~/.julia/packages/Makie/umL6V/src/display.jl:59
 [22] #display#886
    @ ~/.julia/packages/Makie/umL6V/src/display.jl:53 [inlined]
 [23] display
    @ ~/.julia/packages/Makie/umL6V/src/display.jl:53 [inlined]
 [24] #display#885
    @ ~/.julia/packages/Makie/umL6V/src/display.jl:52 [inlined]
 [25] display(fap::Makie.FigureAxisPlot)
    @ Makie ~/.julia/packages/Makie/umL6V/src/display.jl:52
 [26] #invokelatest#2
    @ ./essentials.jl:716 [inlined]
 [27] invokelatest
    @ ./essentials.jl:714 [inlined]
 [28] print_response(errio::IO, response::Any, show_value::Bool, have_color::Bool, specialdisplay::Union{Nothing, AbstractDisplay})
    @ REPL ~/binaries/julia-1.7.2/share/julia/stdlib/v1.7/REPL/src/REPL.jl:293
 [29] (::REPL.var"#45#46"{REPL.LineEditREPL, Pair{Any, Bool}, Bool, Bool})(io::Any)
    @ REPL ~/binaries/julia-1.7.2/share/julia/stdlib/v1.7/REPL/src/REPL.jl:277
 [30] with_repl_linfo(f::Any, repl::REPL.LineEditREPL)
    @ REPL ~/binaries/julia-1.7.2/share/julia/stdlib/v1.7/REPL/src/REPL.jl:510
 [31] print_response(repl::REPL.AbstractREPL, response::Any, show_value::Bool, have_color::Bool)
    @ REPL ~/binaries/julia-1.7.2/share/julia/stdlib/v1.7/REPL/src/REPL.jl:275
 [32] (::REPL.var"#do_respond#66"{Bool, Bool, REPL.var"#77#87"{REPL.LineEditREPL, REPL.REPLHistoryProvider}, REPL.LineEditREPL, REPL.LineEdit.Prompt})(s::REPL.LineEdit.MIState, buf::Any, ok::Bool)
    @ REPL ~/binaries/julia-1.7.2/share/julia/stdlib/v1.7/REPL/src/REPL.jl:846
 [33] #invokelatest#2
    @ ./essentials.jl:716 [inlined]
 [34] invokelatest
    @ ./essentials.jl:714 [inlined]
 [35] run_interface(terminal::REPL.Terminals.TextTerminal, m::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
    @ REPL.LineEdit ~/binaries/julia-1.7.2/share/julia/stdlib/v1.7/REPL/src/LineEdit.jl:2493
 [36] run_frontend(repl::REPL.LineEditREPL, backend::REPL.REPLBackendRef)
    @ REPL ~/binaries/julia-1.7.2/share/julia/stdlib/v1.7/REPL/src/REPL.jl:1232
 [37] (::REPL.var"#49#54"{REPL.LineEditREPL, REPL.REPLBackendRef})()
    @ REPL ./task.jl:423

(MakieSlides) pkg> st
     Project MakieSlides v0.1.0
      Status `~/wd/MakieSlides.jl/Project.toml`
  [159f3aea] Cairo v1.0.5
  [13f3f980] CairoMakie v0.8.3
  [5ae59095] Colors v0.12.8
  [a80b9123] CommonMark v0.8.6
  [5789e2e9] FileIO v1.14.0
  [e9467ef8] GLMakie v0.6.3
  [82e4d734] ImageIO v0.6.5
  [682c06a0] JSON v0.21.3
  [ee78f7c6] Makie v0.17.3
  [20f20a25] MakieCore v0.3.1
  [9b87118b] PackageCompiler v2.0.6
  [438e738f] PyCall v1.93.1
  [a5390f91] ZipFile v0.9.4
  [d6f4376e] Markdown
  [de0858da] Printf
```
</details>

I am waiting for the next release of `Makie` to upgrade so that we can also get #12 merged.

Reply:
Strange.  Do you know if MakieTeX v0.2.3 works on your machine?  That might narrow down the source of the problem Just try the code:
```julia
using Makie, MakieTeX
using CairoMakie # or whichever other backend
fig = Figure()
l1 = Label(
    fig[1, 1], L"A \emph{convex} function $f \in C$ is \textcolor{blue}{denoted} as \tikz{\draw[line width=1pt, >->] (0, -2pt) arc (-180:0:8pt);}";
    tellwidth = false, tellheight = true
)
ax1 = Axis(
    fig[2, 1];
    xtickformat = x -> latexstring.("a_{" .* string.(x) .* "}"),
    ytickformat = x -> latexstring.(string.(x)),
    ylabel = L"\displaystyle \Phi(\vec x) = f(\vec x) + g(V)",
)
heatmap!(ax1, Makie.peaks())
fig
```

Reply:
Indeed, my MWE does work with the current master of Makie.

Trying your MWE with `MakieTeX` in a fresh environment gives me a different error that seems to be related to my `LaTeX` installation:
<details>
<summary>Stack trace</summary>
```julia
julia> include("mwe.jl")
Latex did not write temp.pdf!  Using the `lualatex` engine.
Files in temp directory are:
temp.aux,temp.fdb_latexmk,temp.fls,temp.log,temp.tex
Stdout
Rc files read:
  /etc/latexmkrc
Latexmk: This is Latexmk, John Collins, 17 Mar. 2022. Version 4.77, version: 4.77.
Latexmk: Changing directory to './'
Latexmk: applying rule 'lualatex'...
Rule 'lualatex': File changes, etc:
   Changed files, or newly in use since previous run(s):
  temp.tex
Rule 'lualatex': The following rules & subrules became out-of-date:
  lualatex
------------
Run number 1 of rule 'lualatex'
------------
------------
Running 'lualatex  -file-line-error -halt-on-error --shell-escape -interaction=nonstopmode -recorder  "temp.tex"'
------------
This is LuaHBTeX, Version 1.13.0 (TeX Live 2021)
 system commands enabled.
(./temp.tex
LaTeX2e <2020-10-01> patch level 4
 L3 programming layer <2021-05-07>
(/usr/share/texlive/texmf-dist/tex/generic/luatex85/luatex85.sty)

! LaTeX Error: File `standalone.cls' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: cls)

Enter file name:
./temp.tex:4: Emergency stop.
<read *>

l.4

 301 words of node memory still in use:
   1 hlist, 1 dir, 3 kern, 1 glyph, 1 attribute, 39 glue_spec, 1 attribute_list
 nodes
   avail lists: 2:10,3:3,4:1
./temp.tex:4:  ==> Fatal error occurred, no output PDF file produced!
Transcript written on temp.log.
Latexmk: Getting log file 'temp.log'
Latexmk: Examining 'temp.fls'
Latexmk: Examining 'temp.log'
Latexmk: Errors, so I did not complete making targets
Collected error summary (may duplicate other messages):
  lualatex: Command for 'lualatex' gave return code 1
      Refer to 'temp.log' for details
Latexmk: Undoing directory change

Stderr
Latexmk: Missing input file 'standalone.cls' (or dependence on it) from following:
  '! LaTeX Error: File `standalone.cls' not found.'
Latexmk: If appropriate, the -f option can be used to get latexmk
  to try to force complete processing.

ERROR: LoadError: failed process: Process(`/home/florian/.julia/artifacts/07ef9c18296d78f06f5b946ed2c127b4aae6ac2c/bin/perl /home/florian/.julia/packages/MakieTeX/9PRQp/src/pdfcrop.pl --margin '0 0 0 0' --gscmd /home/florian/.julia/artifacts/34b31c282aec07497f5c4144cfb3c7f6c2d1a7e0/bin/gs temp.pdf temp_cropped.pdf`, ProcessExited(2)) [2]

Stacktrace:
  [1] pipeline_error
    @ ./process.jl:531 [inlined]
  [2] run(::Cmd; wait::Bool)
    @ Base ./process.jl:446
  [3] run
    @ ./process.jl:444 [inlined]
  [4] (::MakieTeX.var"#16#23"{String, String, String})(perl_exe::String)
    @ MakieTeX ~/.julia/packages/MakieTeX/9PRQp/src/rendering.jl:83
  [5] (::JLLWrappers.var"#2#3"{MakieTeX.var"#16#23"{String, String, String}, String})()
    @ JLLWrappers ~/.julia/packages/JLLWrappers/QpMQW/src/runtime.jl:49
  [6] withenv(::JLLWrappers.var"#2#3"{MakieTeX.var"#16#23"{String, String, String}, String}, ::Pair{String, String}, ::Vararg{Pair{String, String}})
    @ Base ./env.jl:172
  [7] withenv_executable_wrapper(f::Function, executable_path::String, PATH::String, LIBPATH::String, adjust_PATH::Bool, adjust_LIBPATH::Bool)
    @ JLLWrappers ~/.julia/packages/JLLWrappers/QpMQW/src/runtime.jl:48
  [8] invokelatest(::Any, ::Any, ::Vararg{Any}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Base ./essentials.jl:716
  [9] invokelatest(::Any, ::Any, ::Vararg{Any})
    @ Base ./essentials.jl:714
 [10] perl(f::Function; adjust_PATH::Bool, adjust_LIBPATH::Bool)
    @ Perl_jll ~/.julia/packages/JLLWrappers/QpMQW/src/products/executable_generators.jl:21
 [11] perl(f::Function)
    @ Perl_jll ~/.julia/packages/JLLWrappers/QpMQW/src/products/executable_generators.jl:21
 [12] mtperl
    @ ~/.julia/packages/MakieTeX/9PRQp/src/rendering.jl:13 [inlined]
 [13] (::MakieTeX.var"#15#22"{String, String})(gs_exe::String)
    @ MakieTeX ~/.julia/packages/MakieTeX/9PRQp/src/rendering.jl:82
 [14] (::JLLWrappers.var"#2#3"{MakieTeX.var"#15#22"{String, String}, String})()
    @ JLLWrappers ~/.julia/packages/JLLWrappers/QpMQW/src/runtime.jl:49
 [15] withenv(::JLLWrappers.var"#2#3"{MakieTeX.var"#15#22"{String, String}, String}, ::Pair{String, String}, ::Vararg{Pair{String, String}})
    @ Base ./env.jl:172
 [16] withenv_executable_wrapper(f::Function, executable_path::String, PATH::String, LIBPATH::String, adjust_PATH::Bool, adjust_LIBPATH::Bool)
    @ JLLWrappers ~/.julia/packages/JLLWrappers/QpMQW/src/runtime.jl:48
 [17] invokelatest(::Any, ::Any, ::Vararg{Any}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Base ./essentials.jl:716
 [18] invokelatest(::Any, ::Any, ::Vararg{Any})
    @ Base ./essentials.jl:714
 [19] gs(f::Function; adjust_PATH::Bool, adjust_LIBPATH::Bool)
    @ Ghostscript_jll ~/.julia/packages/JLLWrappers/QpMQW/src/products/executable_generators.jl:21
 [20] gs(f::Function)
    @ Ghostscript_jll ~/.julia/packages/JLLWrappers/QpMQW/src/products/executable_generators.jl:21
 [21] #14
    @ ~/.julia/packages/MakieTeX/9PRQp/src/rendering.jl:81 [inlined]
 [22] (::Base.RedirectStdStream)(thunk::MakieTeX.var"#14#21"{String, String}, stream::Base.DevNull)
    @ Base ./stream.jl:1359
 [23] #13
    @ ~/.julia/packages/MakieTeX/9PRQp/src/rendering.jl:80 [inlined]
 [24] (::Base.RedirectStdStream)(thunk::MakieTeX.var"#13#20"{String, String}, stream::Base.DevNull)
    @ Base ./stream.jl:1359
 [25] (::MakieTeX.var"#11#18"{Cmd, Cmd, String, Cmd})()
    @ MakieTeX ~/.julia/packages/MakieTeX/9PRQp/src/rendering.jl:79
 [26] cd(f::MakieTeX.var"#11#18"{Cmd, Cmd, String, Cmd}, dir::String)
    @ Base.Filesystem ./file.jl:110
 [27] #10
    @ ~/.julia/packages/MakieTeX/9PRQp/src/rendering.jl:30 [inlined]
 [28] mktempdir(fn::MakieTeX.var"#10#17"{Cmd, Cmd, String, Cmd}, parent::String; prefix::String)
    @ Base.Filesystem ./file.jl:750
 [29] mktempdir (repeats 2 times)
    @ ./file.jl:748 [inlined]
 [30] #compile_latex#9
    @ ~/.julia/packages/MakieTeX/9PRQp/src/rendering.jl:29 [inlined]
 [31] compile_latex
    @ ~/.julia/packages/MakieTeX/9PRQp/src/rendering.jl:23 [inlined]
 [32] #latex2pdf#25
    @ ~/.julia/packages/MakieTeX/9PRQp/src/rendering.jl:102 [inlined]
 [33] latex2pdf
    @ ~/.julia/packages/MakieTeX/9PRQp/src/rendering.jl:102 [inlined]
 [34] CachedTeX(doc::TeXDocument; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ MakieTeX ~/.julia/packages/MakieTeX/9PRQp/src/types.jl:106
 [35] CachedTeX
    @ ~/.julia/packages/MakieTeX/9PRQp/src/types.jl:106 [inlined]
 [36] to_plottable_cachedtex
    @ ~/.julia/packages/MakieTeX/9PRQp/src/text_override.jl:61 [inlined]
 [37] (::MakieTeX.var"#63#64")(ltex::LaTeXString, font::FreeTypeAbstraction.FTFont, textsize::Float32, lineheight::Float32, color::ColorTypes.RGBA{Float32})
    @ MakieTeX ~/.julia/packages/MakieTeX/9PRQp/src/text_override.jl:208
 [38] map(::MakieTeX.var"#63#64", ::Observable{LaTeXString}, ::Observable{Any}, ::Vararg{Observable{Any}}; ignore_equal_values::Bool)
    @ Observables ~/.julia/packages/Observables/ynr7h/src/Observables.jl:444
 [39] map(::MakieTeX.var"#63#64", ::Observable{LaTeXString}, ::Observable{Any}, ::Observable{Any}, ::Vararg{Observable{Any}})
    @ Observables ~/.julia/packages/Observables/ynr7h/src/Observables.jl:444
 [40] plot!(t::MakieCore.Text{Tuple{LaTeXString}})
    @ MakieTeX ~/.julia/packages/MakieTeX/9PRQp/src/text_override.jl:207
 [41] plot!(scene::Scene, P::Type{MakieCore.Text{Tuple{LaTeXString}}}, attributes::Attributes, input::Tuple{Observable{Any}}, args::Observable{Tuple{LaTeXString}})
    @ Makie ~/.julia/packages/Makie/fEZv2/src/interfaces.jl:402
 [42] plot!(scene::Scene, P::Type{MakieCore.Text}, attributes::Attributes, args::Observable{Any}; kw_attributes::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Makie ~/.julia/packages/Makie/fEZv2/src/interfaces.jl:320
 [43] plot!
    @ ~/.julia/packages/Makie/fEZv2/src/interfaces.jl:288 [inlined]
 [44] #plot!#137
    @ ~/.julia/packages/Makie/fEZv2/src/interfaces.jl:271 [inlined]
 [45] #text!#52
    @ ~/.julia/packages/MakieCore/oBlaS/src/recipes.jl:37 [inlined]
 [46] initialize_block!(l::Label)
    @ Makie.MakieLayout ~/.julia/packages/Makie/fEZv2/src/makielayout/blocks/label.jl:11
 [47] _block(::Type{Label}, ::Figure; bbox::Nothing, kwargs::Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol, Symbol}, NamedTuple{(:text, :tellwidth, :tellheight), Tuple{LaTeXString, Bool, Bool}}})
    @ Makie.MakieLayout ~/.julia/packages/Makie/fEZv2/src/makielayout/blocks.jl:395
 [48] _block(::Type{Label}, ::GridPosition; kwargs::Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol, Symbol}, NamedTuple{(:text, :tellwidth, :tellheight), Tuple{LaTeXString, Bool, Bool}}})
    @ Makie.MakieLayout ~/.julia/packages/Makie/fEZv2/src/makielayout/blocks.jl:284
 [49] #_#38
    @ ~/.julia/packages/Makie/fEZv2/src/makielayout/blocks.jl:269 [inlined]
 [50] #Label#552
    @ ~/.julia/packages/Makie/fEZv2/src/makielayout/blocks/label.jl:1 [inlined]
 [51] top-level scope
    @ ~/wd/scratch/dummy/mwe.jl:4
 [52] include(fname::String)
    @ Base.MainInclude ./client.jl:451
in expression starting at /home/florian/wd/scratch/dummy/mwe.jl:4

caused by:
Stacktrace:
  [1] error()
    @ Base ./error.jl:42
  [2] (::MakieTeX.var"#11#18"{Cmd, Cmd, String, Cmd})()
    @ MakieTeX ~/.julia/packages/MakieTeX/9PRQp/src/rendering.jl:62
  [3] cd(f::MakieTeX.var"#11#18"{Cmd, Cmd, String, Cmd}, dir::String)
    @ Base.Filesystem ./file.jl:110
  [4] #10
    @ ~/.julia/packages/MakieTeX/9PRQp/src/rendering.jl:30 [inlined]
  [5] mktempdir(fn::MakieTeX.var"#10#17"{Cmd, Cmd, String, Cmd}, parent::String; prefix::String)
    @ Base.Filesystem ./file.jl:750
  [6] mktempdir (repeats 2 times)
    @ ./file.jl:748 [inlined]
  [7] #compile_latex#9
    @ ~/.julia/packages/MakieTeX/9PRQp/src/rendering.jl:29 [inlined]
  [8] compile_latex
    @ ~/.julia/packages/MakieTeX/9PRQp/src/rendering.jl:23 [inlined]
  [9] #latex2pdf#25
    @ ~/.julia/packages/MakieTeX/9PRQp/src/rendering.jl:102 [inlined]
 [10] latex2pdf
    @ ~/.julia/packages/MakieTeX/9PRQp/src/rendering.jl:102 [inlined]
 [11] CachedTeX(doc::TeXDocument; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ MakieTeX ~/.julia/packages/MakieTeX/9PRQp/src/types.jl:106
 [12] CachedTeX
    @ ~/.julia/packages/MakieTeX/9PRQp/src/types.jl:106 [inlined]
 [13] to_plottable_cachedtex
    @ ~/.julia/packages/MakieTeX/9PRQp/src/text_override.jl:61 [inlined]
 [14] (::MakieTeX.var"#63#64")(ltex::LaTeXString, font::FreeTypeAbstraction.FTFont, textsize::Float32, lineheight::Float32, color::ColorTypes.RGBA{Float32})
    @ MakieTeX ~/.julia/packages/MakieTeX/9PRQp/src/text_override.jl:208
 [15] map(::MakieTeX.var"#63#64", ::Observable{LaTeXString}, ::Observable{Any}, ::Vararg{Observable{Any}}; ignore_equal_values::Bool)
    @ Observables ~/.julia/packages/Observables/ynr7h/src/Observables.jl:444
 [16] map(::MakieTeX.var"#63#64", ::Observable{LaTeXString}, ::Observable{Any}, ::Observable{Any}, ::Vararg{Observable{Any}})
    @ Observables ~/.julia/packages/Observables/ynr7h/src/Observables.jl:444
 [17] plot!(t::MakieCore.Text{Tuple{LaTeXString}})
    @ MakieTeX ~/.julia/packages/MakieTeX/9PRQp/src/text_override.jl:207
 [18] plot!(scene::Scene, P::Type{MakieCore.Text{Tuple{LaTeXString}}}, attributes::Attributes, input::Tuple{Observable{Any}}, args::Observable{Tuple{LaTeXString}})
    @ Makie ~/.julia/packages/Makie/fEZv2/src/interfaces.jl:402
 [19] plot!(scene::Scene, P::Type{MakieCore.Text}, attributes::Attributes, args::Observable{Any}; kw_attributes::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Makie ~/.julia/packages/Makie/fEZv2/src/interfaces.jl:320
 [20] plot!
    @ ~/.julia/packages/Makie/fEZv2/src/interfaces.jl:288 [inlined]
 [21] #plot!#137
    @ ~/.julia/packages/Makie/fEZv2/src/interfaces.jl:271 [inlined]
 [22] #text!#52
    @ ~/.julia/packages/MakieCore/oBlaS/src/recipes.jl:37 [inlined]
 [23] initialize_block!(l::Label)
    @ Makie.MakieLayout ~/.julia/packages/Makie/fEZv2/src/makielayout/blocks/label.jl:11
 [24] _block(::Type{Label}, ::Figure; bbox::Nothing, kwargs::Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol, Symbol}, NamedTuple{(:text, :tellwidth, :tellheight), Tuple{LaTeXString, Bool, Bool}}})
    @ Makie.MakieLayout ~/.julia/packages/Makie/fEZv2/src/makielayout/blocks.jl:395
 [25] _block(::Type{Label}, ::GridPosition; kwargs::Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol, Symbol}, NamedTuple{(:text, :tellwidth, :tellheight), Tuple{LaTeXString, Bool, Bool}}})
    @ Makie.MakieLayout ~/.julia/packages/Makie/fEZv2/src/makielayout/blocks.jl:284
 [26] #_#38
    @ ~/.julia/packages/Makie/fEZv2/src/makielayout/blocks.jl:269 [inlined]
 [27] #Label#552
    @ ~/.julia/packages/Makie/fEZv2/src/makielayout/blocks/label.jl:1 [inlined]
 [28] top-level scope
    @ ~/wd/scratch/dummy/mwe.jl:4
 [29] include(fname::String)
    @ Base.MainInclude ./client.jl:451
 [30] top-level scope
    @ REPL[6]:1
```
</details>

Perhaps we should just upgrade now, because we have to it anyways.

Reply:
Huh...strange!  Yeah, I guess it would be easier to just build off Makie master.  Not sure what changed though.

For the MakieTeX thing you could try ```  MakieTeX.CURRENT_TEX_ENGINE[] = `tectonic` ```.  It looks like luatex is accessible on your system but you don't have the `standalone` documentclass installed, possibly you installed a minimal tex live.

Reply:
I upgraded to `v0.17.4` and the overflow error is gone.
I also upgrade my `texlive` installation on Fedora Linux to include all packages (~3.1Gs) and the `MakieTeX` example is now working too.

Reply:
I incorporated your suggestion and now I am almost there.